src/tests/test_forcefield.py

```diff
@@ -16,33 +14,30 @@ class FFTests(object):
 
     def test_FF_yields_consistent_results(self):
         """Check whether multiple calls to FF yield the same result"""
-        self.logger.debug("\nChecking consistency of ForceField constructor\n")
-        self.assertEqual(forcefield.FF(self.options),forcefield.FF(self.options),
-        msg = "\nGot two different forcefields despite using the same options as input")
+        print("\nChecking consistency of ForceField constructor\n")
+        assert forcefield.FF(self.options) == forcefield.FF(self.options), "Got two different forcefields despite using the same options as input"
 
     def test_make_function_return_value(self):
         """Check that make() return value meets expectation"""
         pvals = self.ff.pvals0
 
-        self.logger.debug("Running forcefield.make() with zero vector should not change pvals... ")
+        print("Running forcefield.make() with zero vector should not change pvals... ")
         new_pvals = np.array(self.ff.make(np.zeros(self.ff.np)))
-        self.assertEqual(pvals.size,new_pvals.size)
-        # given zero matrix, make should return unchanged pvals
-        self.assertEqual(pvals,new_pvals,
-                    msg="\nmake() should produce unchanged pvals when given zero vector")
-        self.logger.debug("ok\n")
+        assert pvals.size == new_pvals.size
+        assert (pvals == new_pvals).all(), "make() should produce unchanged pvals when given zero vector"
+        print("ok\n")
 
-        self.logger.debug("make() should return different values when passed in nonzero pval matrix... ")
+        print("make() should return different values when passed in nonzero pval matrix... ")
         new_pvals = np.array(self.ff.make(np.ones(self.ff.np)))
-        self.assertEqual(pvals.size,new_pvals.size)
+        assert pvals.size == new_pvals.size
         # given arbitrary nonzero input, make should return new pvals
-        self.assertFalse((pvals==new_pvals).all(), msg="\nmake() returned unchanged pvals even when given nonzero matrix")
-        self.logger.debug("ok\n")
+        assert not (pvals==new_pvals).all(), "make() returned unchanged pvals even when given nonzero matrix"
+        print("ok\n")
 
-        self.logger.debug("make(use_pvals=True) should return the same pvals... ")
+        print("make(use_pvals=True) should return the same pvals... ")
         new_pvals = np.array(self.ff.make(np.ones(self.ff.np),use_pvals=True))
-        self.assertEqual(np.ones(self.ff.np),new_pvals, msg="\nmake() did not return input pvals with use_pvals=True")
-        self.logger.debug("ok\n")
+        assert (np.ones(self.ff.np) == new_pvals).all(), "make() did not return input pvals with use_pvals=True"
```

Comment:
`==` may not be an appropriate replacement for all uses of `self.assertEqual`, since `unittest.assertEqual` has its behavior overridden specifically for numpy arrays in `__init__`. So we should use something like Hyesu's 
`np.testing.assert_array_almost_equal(EXPECTED_WATER_RESULTS,result,decimal=0.001, err_msg=msg) `
below to explicitly describe how we're doing `np.array` comparisons (and with what tolerance)

Reply:
Yes, I think that maybe it's best to choose the proper comparison function or operator on a case-by-case basis rather than defining a catch-all comparison function and trying to use it everywhere.